src/mongocxx/collection.cpp

```diff
@@ -363,69 +357,88 @@ stdx::optional<bsoncxx::document::value> collection::find_one_and_replace(
     return b.extract();
 }
 
+stdx::optional<bsoncxx::document::value> collection::find_one_and_replace(
+    bsoncxx::document::view filter, bsoncxx::document::view replacement,
+    const options::find_one_and_replace& options) {
+    auto opts = libmongoc::find_and_modify_opts_new();
```

Comment:
capture the result of find_and_modify_opts_new into a unique_ptr where the deleter is ibmongoc::find_and_modify_opts_destroy